### DISCUSSION TITLE
Weird Subfolders

### DISCUSSION THREAD

**Discussion by trleith** [automatic-ripping-machine/automatic-ripping-machine#847] [General]
Sometimes ARM makes odd subfolders that can go several layers deep. I always find a perfectly good FLAC down at the bottom. It seems to put cover art down there if the medley was the last track. Otherwise you get "albumart_backup" instead of putting the art in the top level where it goes. But it's inconsistent. That'll be the next thing I look into.

Investigate Investigate Investigate Investigate Investigate Investigate .....

OK, any time a disc has a medley it seems to do this. You get a randomly-named subdirectory per selection in the medley which is entirely empty except for the next subdirectory. At the bottom, you get one perfectly good FLAC that has all the music for the whole medley. The track has the name of the last selection of the medley, and no track number.

Since Plex is completely brain-dead about anything except a "song" that outcome where the medley is "merged" into a single FLAC isn't completely terrible from my point of view, except what I really need is the FLAC moved to the top level, given the proper track number, and given the name of the **_medley_** as the track title and filename. Bonus points if it could somehow indicate that it is a medley. 

In this particular case, it's Track 7 Weather Report Suite that should have three "track markers" in it: Prelude / Part 1 / Part 2 - Let it Grow 

(Given the limitations of Plex) What I think I want in my top level is:

01 - Mississippi Half-Step Uptown Toodleoo.flac
02 - Let Me Sing Your Blues Away.flac
03 - Row Jimmy.flac
04 - Stella Blue.flac
05 - Here Comes Sunshine.flac
06 - Eyes of the World.flac
07 - Weather Report Suite (medley).flac

Any temporary directories it had to create should be deleted.

If there's a CD ripper that can fully-utilize the FLAC Cue Sheet / Index features and a tagger to get all the metadata tagging from (say) MusicBrainz correct, and all that, I'd love to know about it. Then I need some kind of FLAC splitter to make brain-dead media servers happy. But the data in MusicBrainz has wildly-varying quality, so...

Sometimes I think I'd be better off playing physical CDs. But I find I listen to more of my collection this way. <sigh>

**Comment by trleith** [comment]
So here I am commenting on my own topic.

It seems the internal tags in the FLACs were correct; ARM it got something reasonable from MusicBrainz. This leaves the subdirectory and file naming issue. I note in both cases the MusicBrainz track title had forward slashes between the "selections" in the medleys.

I want to experiment with ripping to one big FLAC with an internal cuesheet. I think Plex will choke on it. I wonder whether Plex would work better with FLAC encoding in an MKV container... too much to hope for, eh?

I used Picard to fix it up.

               t

`
05/22/2023  05:48 PM        33,234,753 01 - Mississippi Half-Step Uptown Toodleoo.flac

05/22/2023  05:48 PM        20,298,481 02 - Let Me Sing Your Blues Away.flac

05/22/2023  05:48 PM        38,631,574 03 - Row Jimmy.flac

05/22/2023  05:48 PM        32,419,092 04 - Stella Blue.flac

05/22/2023  05:48 PM        27,103,747 05 - Here Comes Sunshine.flac

05/22/2023  05:48 PM        30,494,867 06 - Eyes of the World.flac

05/22/2023  05:48 PM        73,017,344 07 - Weather Report Suite_ Prelude _ Part 1 _ Part 2_ Let It Grow.flac

05/22/2023  10:26 AM           126,654 cover.jpg

               8 File(s)    255,326,512 bytes

`

**Reply by microtechno9000** [reply]
Have you modified the .abcde config file? or using the default that ships with ARM?

When you say there are strange folders, are they in Music or in a .abcde folder?

**Reply by trleith** [reply]
Oh, I see more

Track 7 is a medley. In MusicBrainz, the track title is `Weather Report Suite: Prelude / Part 1 / Part 2: Let It Grow `

Yes, with the forward slashes

This is how it gets parsed-out by ARM/abcde

Looked at from the Linux side:

> trl@svr2:/plex/arm/music/Grateful Dead/Wake of the Flood$ ls -la
> total 177272
> drwxr-xr-x 3 arm arm     4096 May 23 00:00  .
> drwxr-xr-x 5 arm arm     4096 May 22 23:57  ..
> -rw-r--r-- 1 arm arm 33118909 May 22 23:58 '01 - Mississippi Half-Step Uptown Toodleoo.flac'
> -rw-r--r-- 1 arm arm 20195570 May 22 23:58 '02 - Let Me Sing Your Blues Away.flac'
> -rw-r--r-- 1 arm arm 38510366 May 22 23:59 '03 - Row Jimmy.flac'
> -rw-r--r-- 1 arm arm 32304088 May 22 23:59 '04 - Stella Blue.flac'
> -rw-r--r-- 1 arm arm 26994045 May 23 00:00 '05 - Here Comes Sunshine.flac'
> -rw-r--r-- 1 arm arm 30381745 May 23 00:00 '06 - Eyes of the World.flac'
> drwxr-xr-x 3 arm arm     4096 May 22 23:57 '07 - Weather Report Suite - Prelude '
> 

From Windows' point of view, the directory name _07 - Weather Report Suite - Prelude_ is illegal, so it shows up in Windows Explorer as 0EGI6~Z. But it has its own subdirectory ' Part 1'

Pardon the art -- I haven't got this comment system figured out yet.

>Grateful Dead
>|
>+=== Wake of the Flood
>|............... Tracks 1 - 6 as usual
>+======= 0EGI6\~Z or 07 - Weather Report Suite - Prelude
>........................|
>.......................+===  _DVCYT\~R or ' Part 1' (notice the leading space in the directory name)
>.......................|..............cover.jpg
>.......................|................' Part 2 - Let It Grow.flac'  (notice the leading space in the filename. On the Windows side, it's gone)
>

metaflac --list ' Part 2 - Let It Grow.flac'

(partial output)

> METADATA block #2
>   type: 4 (VORBIS_COMMENT)
>   is last: false
>   length: 226
>   vendor string: reference libFLAC 1.3.3 20190804
>   comments: 7
>     comment[0]: ARTIST=Grateful Dead
>     comment[1]: ALBUM=Wake of the Flood
>     comment[2]: TITLE=Weather Report Suite: Prelude / Part 1 / Part 2: Let It Grow
>     comment[3]: DATE=1995
>     comment[4]: TRACKNUMBER=07
>     comment[5]: TRACKTOTAL=07
>     comment[6]: CDDB=6c0aa807
> METADATA block #3
>   type: 1 (PADDING)
>   is last: true
>   length: 8006
